src/services/firebase.ts

```
@@ -350,6 +432,225 @@
+  cacheHits: number;
+  lastQueryTime: number;
+  avgLatency: number;
+  p95Latency: number; // 95th percentile latency tracking
+  connectionPoolHits: number;
+  networkErrors: number;
+  latencyHistory: number[]; // Keep last 20 latencies for percentile calculation
+}
+
+const queryMetrics = new Map<string, QueryMetrics>();
+
+function updateQueryMetrics(queryKey: string, latency: number, fromCache: boolean, networkError: boolean = false): void {
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)